### PR TITLE
Change to mofed to version 5.2-0.5.7.0 and uninstall rdma-core

### PR DESCRIPTION
**Pull request by abdallahyas**
Changed the mofed image version to 5.2-0.5.7.0 from 5.0 and made
the necessary changes to make it work by uninstalling the rdma-core
package for any test that runs the ofed, and install it for tests
that run rdma tests only.

### CODE REVIEW

File: common/common_functions.sh [hunk 1]
```diff
@@ -355,19 +355,38 @@ function deploy_calico {
 }
 
 function create_macvlan_net {
-    local macvlan_file="${ARTIFACTS}/macvlan-net.yaml"
+    local macvlan_file="${1:-$ARTIFACTS/macvlan-net.yaml}"
```
Comment (adrianchiris) [inline common/common_functions.sh]: are the changes in create_macvlan_net related to the PR ?

its not mentioned in the commit message

Comment (abdallahyas) [reply common/common_functions.sh]: @adrianchiris This will be removed in favor of the udev rule

**Comment by adrianchiris** [comment]
@AbdYsn can this be merged once CI passes ?

**Comment by abdallahyas** [comment]
@adrianchiris yes, but it is expected for it to fail because of Mellanox/network-operator#66 (which is the cause of the daily failure), so i think this change should depend on that patch.

**Comment by abdallahyas** [comment]
@adrianchiris This is ready

**Review by adrianchiris — APPROVED** [review]
We should really avoid changing the test environment by loading/unloading modules or installing/uninstalling packages in the test internals or in helper functions that should really do one thing e.g `configure_ofed` that should just configure MOFED in the nicclusterpolicy CR.

approving this PR but lets followup with this refactor to separate test environment setup/cleanup from the test itself.

**Pull request merged**